packages/vscode-extension/src/debug/taskTerminal/localTunnelTaskTerminal.ts

```diff
@@ -195,19 +202,80 @@ export class LocalTunnelTaskTerminal extends BaseTaskTerminal {
           );
         }
       });
+
+      setTimeout(() => {
+        if (!this.status.endpoint) {
+          this.saveNgrokEndpointFromApi(configFile, tunnelName).then((res) => {
+            if (res.isErr()) {
+              resolve(res);
+            }
+          });
+        }
+      }, ngrokTimeout);
     });
   }
 
-  private command(configFile: string) {
-    return `ngrok start ${ngrokTunnelName} --config=${configFile} --log=stdout --log-format=logfmt`;
+  private command(configFile: string, tunnelName: string) {
+    return `ngrok start ${tunnelName} --config=${configFile} --log=stdout --log-format=logfmt`;
   }
 
-  private parseNgrokEndpointFromLog(data: string): EndpointInfo | undefined {
-    const matches = data.match(ngrokEndpointRegex);
+  private saveNgrokEndpointFromLog(data: string, tunnelName: string): void {
+    const matches = data.match(ngrokEndpointRegex(tunnelName));
     if (matches && matches?.length > 2) {
-      return { src: matches[1], dist: matches[2] };
+      const ngrokTunnelInfo = { src: matches[1], dist: matches[2] };
+      this.isOutputSummary = true;
+      this.status.endpoint = ngrokTunnelInfo;
+      this.outputSuccessSummary(ngrokTunnelInfo);
+    }
+  }
+
+  private async saveNgrokEndpointFromApi(
```

Comment:
Which exact case that log fails but api works?

Reply:
We use regex `obj=tunnels name=${tunnelName} addr=(?<src>.*) url=(?<endpoint>.*)` to find the endpoint. 

The log may be changed in some version of ngrok, (Use global ngrok .......)